content/documentation/pages/6-recipes/7-scaling/2-autoscaling.md

```diff
@@ -0,0 +1,204 @@
+
+<!--TIP-->
+
+When using RabbitMQ, the `app.log.spring.cloud.stream.kafka.binder.` deployment properties are not required.
+
+<!--END_TIP-->
+
+![SCDF autoscaling stream deploying](images/scdf-autoscaling-stream-deploying.png)
+
+With the help of the SCDF built-in Grafana application dashboard check the [applications throughput metrics](%currentPath%/feature-guides/streams/monitoring/#prometheus-1):
+![SCDF autoscaling initial metrics](images/scdf-autoscaling-initial-metrics.png)
+
+The `time`, `transform` and `log` applications maintain the same message throughput (~ 1 msg/s). The `transform` copes fine with the present load.
+
+### Increase pipeline load
+
+Now, let's increase the load by increasing the time-source's message production rate. By changing time's `time-unit` property from `MILLISECONDS` to `MICROSECONDS` the input rate will increase from one to a couple of thousands of messages per second.
+Note that the [stream update](%currentPath%/stream-developer-guides/continuous-delivery/) functionality allows to update the time application without stopping the entire stream:
+
+```shell
+stream update --name scaletest --properties "app.time.trigger.time-unit=MICROSECONDS"
+```
+
+The `time` app is re-deployed with the new time-unit property:
+![SCDF autoscaling increase load](images/scdf-autoscaling-increase-load.png)
+
+Now `time` source emits messages with a rate of `~5000 msg/s`. The `transform` processor though is capped at around `1000 msg/s` and that in turn, it halts the throughput of the entire stream to a certain level. This is an Indicator that the `transform` has become a bottleneck.
+
+![SCDF autoscaling pipeline bottleneck](images/scdf-autoscaling-pipeline-bottleneck.png)
+
+The `HighThroughputDifference` Prometheus alert rule detects the rate discrepancy and fires a scale-out alert:
+
+![SCDF autoscaling scaleout alert](images/scdf-autoscaling-scaleout-alert.png)
+
+in result 3 additional transform instances are added:
+
+![SCDF autoscaling adding 3 instances](images/scdf-autoscaling-adding-3-instances.png)
+
+with the help of the additional instance the `log` sink and the entire pipeline catch up match with the `time` source:
+
+![SCDF autoscaling stream catches up](images/scdf-autoscaling-stream-catchup.png)
+
+### Reduce pipeline load
+
+If we reduce the input load back to the original rate 1 msg/s):
+
+```shell
+stream update --name scaletest --properties "app.time.trigger.time-unit=MILLISECONDS"
+```
+
+the multiple `transform` instances gradually become redundant which triggers the `ZeroThroughputDifference` alert is fired:
+
+![SCDF autoscaling scale-in alert](images/scdf-autoscaling-scale-in-alert.png)
+
+later triggers the scale-in actions and removes the extra `transform` instance, while the throughput of the entire pipeline is normalized to `~1 msg/s` :
```

Comment:
> With the previous command, the scale-in action removes the extra `transform` instances, and the throughput of the entire pipeline is normalized back to `~1 msg/s` :

Reply:
imo, above sentence is misleading as for autoscaling we do not trigger scale actions explicitly but let the system do it eventually.  
Please check how i changed the last two paragraphs.